src/util-device.c

```diff
@@ -106,6 +109,36 @@ char *LiveGetDeviceName(int number)
 }
 
 /**
+ *  \brief Shorten a device name that is to long
+ *
+ *  \param device name from config and destination for modified
+ *
+ *  \retval None, is added to destination char *newdevname
+ */
+int LiveSafeDeviceName(const char *devname, char *newdevname, size_t destlen)
+{
+    size_t devnamelen = strlen(devname);
+
+    // If we have to shorten the interface name
+    if (devnamelen > MAX_DEVNAME) {
+    
+        // We need 13 chars to do this shortening
+        if (destlen < 13) {
+            return 1;
+        }
+    
+        size_t length;
+        length = strlcpy(newdevname, devname, DEVNAME_CHUNCK);
```

Comment:
The way strlcpy/strlcat are supposed to be used it that you pass the size of the destination buffer as the last argument. This way it can protect against overflows. The pattern here is using the last arg to say something about the input buffer.